prusti-tests/tests/cargo_verify/test_map/src/main.rs

```diff
@@ -0,0 +1,9 @@
+//
+extern crate prusti_contracts;
+use prusti_contracts::*;
+
+fn main() {
+    let m0: Map<u32, u32> = Map::empty();
+    let m1 = m0.insert(0, 0);
+    let m2 = m1.delete(0);
```

Comment:
Although we don't check for this yet, this should be in `ghost! { ... }` as well. In the long term I think we need proper support for ghost arguments and return values but as we discussed with @vakaras and the others today, that's a bit tricky (I had [some ideas](https://hackmd.io/r0kiQvOMTdueQAZxKd4boQ?view#Ghost-arguments-and-extern_spec) but without patching the compiler like `verus` does we can't do that much).